static/shared/js/typeahead.js

```diff
@@ -98,30 +98,13 @@ export function clean_query_lowercase(query) {
     return query;
 }
 
-export const parse_unicode_emoji_code = (code) => {
-    try {
-        return code
-            .split("-")
-            .map((hex) => String.fromCodePoint(Number.parseInt(hex, 16)))
-            .join("");
-    } catch {
-        // Code could not be parsed, but instead of throwing an exception
-        // we return undefined instead of a string.
-        return undefined;
-    }
-};
-
 export function get_emoji_matcher(query) {
     // replaces spaces with underscores for emoji matching
     query = query.replace(/ /g, "_");
     query = clean_query_lowercase(query);
 
     return function (emoji) {
-        const matches_emoji_literal =
-            emoji.emoji_code && parse_unicode_emoji_code(emoji.emoji_code) === query;
```

Comment:
I think concretely, we want `emoji.emoji_type === "unicode_emoji" &&` added to this line?

Reply:
The original implementation of this function only assumed the presence of the `emoji_name` attribute, and nothing else. I can't remember now (should have written it down) but I read in the source code somewhere that it was intentional. You can see evidence of my hesitation with the `emoji.emoji_code &&` check. Adding in another dependency on `emoji.emoji_type` felt like a step too far, and my plan was to re-introduce this logic back in to the mobile app where it originated.

Reply:
I think we do want this behavior for mobile web. We can roll that work into #21778.